pkg/controller/node/cidr_set.go

```diff
@@ -88,24 +120,47 @@ func (s *cidrSet) getBeginingAndEndIndices(cidr *net.IPNet) (begin, end int, err
 	}
 
 	if s.clusterMaskSize < maskSize {
-		subNetMask := net.CIDRMask(s.subNetMaskSize, 32)
-		begin, err = s.getIndexForCIDR(&net.IPNet{
-			IP:   cidr.IP.To4().Mask(subNetMask),
-			Mask: subNetMask,
-		})
-		if err != nil {
-			return -1, -1, err
-		}
-
-		ip := make([]byte, 4)
-		ipInt := binary.BigEndian.Uint32(cidr.IP) | (^binary.BigEndian.Uint32(cidr.Mask))
-		binary.BigEndian.PutUint32(ip, ipInt)
-		end, err = s.getIndexForCIDR(&net.IPNet{
-			IP:   net.IP(ip).To4().Mask(subNetMask),
-			Mask: subNetMask,
-		})
-		if err != nil {
-			return -1, -1, err
+		if cidr.IP.To4() != nil {
```

Comment:
The other difference is the ipv4 code uses Uint32, and the v6 code uses Uint64.
So the single method would probably still need to have an if ipv4 else ipv6 in it. Do you think
it is still worth combining?